charts/velero/templates/upgrade-crds.yaml

```diff
@@ -0,0 +1,48 @@
+{{- if .Release.IsUpgrade }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "velero.fullname" . }}-upgrade-crds
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ include "velero.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ include "velero.chart" . }}
+spec:
+  template:
+    metadata:
+      name: velero-upgrade-crds
+    spec:
+      serviceAccountName: {{ include "velero.serverServiceAccount" . }}
+      initContainers:
+        - name: velero
+      {{- if .Values.image.digest }}
+          image: "{{ .Values.image.repository }}@{{ .Values.image.digest }}"
+      {{- else }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+      {{- end }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          command:
+            - /bin/sh
+            - -c
+            - /velero install --crds-only --dry-run -o yaml > /tmp/crds.yaml
+          volumeMounts:
+            - mountPath: /tmp
+              name: crds
+      containers:
+        - name: kubectl
+          image: docker.io/bitnami/kubectl:1.14.1
```

Comment:
Is there a reason for using this specific version of the `kubectl` image? I'm not totally clear on the version support matrix for kubectl and cluster versions. If possible, it would be preferable to use a newer image, or at least the latest patch release of this one. My concern is that the 1.14.* series was last updated a year ago and could contain security vulnerabilities. 1.16-1.18 have been [deprecated upstream](https://github.com/bitnami/bitnami-docker-kubectl#branch-deprecation-notice) but Bitnami are still making builds of those images for a short time which will include recent security vulnerability patches.

Reply:
Because I choose the same kubectl image as https://github.com/vmware-tanzu/helm-charts/blob/main/charts/velero/templates/cleanup-crd.yaml#L26.

Yes,  I could change to a newer image.

Reply:
I don't have a strong opinion on which kubectl version to use since Velero supports the Kubernetes version from v1.12+.

Reply:
If we're using that image elsewhere, then maybe we can continue using it for now. There might be some risk in upgrading, so we can stick with this for now (but bump to 1.14.3 to incorporate new fixes). We can address upgrading both this usage and the usage in the cleanup template at a later time :+1:

Reply:
I noticed you just upgraded 1.16, if you've tried it out and it works, that's also fine :smile:

Reply:
I tried v1.14.3 on Kubernetes 1.20.5 cluster, it works well.